### PR TITLE
feat: upgrade pyjwt to 2.1.0

### PR DESCRIPTION
**Pull request by iamsobanjaved**
## Description

- Resolved all pyjwt version 2.0 breaking changes
- Overrriden a few global constraints by downloading the common constraint file
- Upgraded edx-drf-extensions to the latest version
- Upgraded edx-auth-backends to the latest version

For more info about pyjwt changes: https://pyjwt.readthedocs.io/en/stable/changelog.html

## Ticket Link

[BOM-2761](https://openedx.atlassian.net/browse/BOM-2761)

## Post-review

Squash commits into discrete sets of changes

### CODE REVIEW

File: enterprise_catalog/apps/catalog/forms.py [hunk 1]
```diff
@@ -24,17 +24,17 @@ class Meta:
         fields = ('content_filter',)
 
     def validate_content_filter_fields(self, content_filter):
-        for key in cftypes:
+        for key, value in cftypes.items():
```
Comment (iamsobanjaved) [inline enterprise_catalog/apps/catalog/forms.py]: These changes have been made to fix quality failures, that arrived due to pylint upgrade

**Review by awais786 — APPROVED** [review]

File: enterprise_catalog/apps/catalog/forms.py (hunk 1 above)
Comment (awais786) [reply enterprise_catalog/apps/catalog/forms.py]: Revert these changes. This PR should contains only jwt changes.

**Comment by iamsobanjaved** [comment]
> ### Testing considerations
> This needs special attention before deployment and testing on the stage of specifically API endpoints with JWT authentication enabled. Similar PRs in credentials, analytics-api, and course-discovery already deployed on production.
> 

@edx/enterprise-titans kindly review and deploy this PR with these testing considerations

**Comment by christopappas** [comment]
I handled the conflicts and rebasing on this separate PR: https://github.com/edx/enterprise-catalog/pull/327 -- going to close this one! Thank you for the contribution :)

**Pull request closed without merging**